src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -87,6 +80,97 @@ private void VerifyTypeIL(CSharpCompilation compilation, string typeName, string
+{
+    public override int P1 { get => field; set => field = value; }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            if (callGetFieldsToEmit)
+            {
+                var baseFields = comp.GetTypeByMetadataName("Base").GetFieldsToEmit().ToArray();
+                Assert.Equal(1, baseFields.Length);
+                Assert.Equal("System.Int32 Base.<P1>k__BackingField", baseFields[0].ToTestDisplayString());
+
+                var derived1Fields = comp.GetTypeByMetadataName("Derived1").GetFieldsToEmit().ToArray();
+                Assert.Equal(2, derived1Fields.Length);
+                Assert.Equal("System.Int32 Derived1.<P1>k__BackingField", derived1Fields[0].ToTestDisplayString());
+                Assert.Equal("System.Int32 Derived1.<P2>k__BackingField", derived1Fields[1].ToTestDisplayString());
+
+                var derived2Fields = comp.GetTypeByMetadataName("Derived2").GetFieldsToEmit().ToArray();
+                Assert.Equal(2, derived2Fields.Length);
+                Assert.Equal("System.Int32 Derived2.<P1>k__BackingField", derived2Fields[0].ToTestDisplayString());
+                Assert.Equal("System.Int32 Derived2.<P2>k__BackingField", derived2Fields[1].ToTestDisplayString());
+
+                var derived3Fields = comp.GetTypeByMetadataName("Derived3").GetFieldsToEmit().ToArray();
+                Assert.Equal(0, derived3Fields.Length);
+
+                var derived4Fields = comp.GetTypeByMetadataName("Derived4").GetFieldsToEmit().ToArray();
+                Assert.Equal(1, derived4Fields.Length);
+                Assert.Equal("System.Int32 Derived4.<P1>k__BackingField", derived4Fields[0].ToTestDisplayString());
+            }
+
+            if (callSemanticModel)
+            {
+                var model = comp.GetSemanticModel(comp.SyntaxTrees.Single());
+                var typeInfo = model.GetTypeInfo(comp.SyntaxTrees.Single().GetRoot().DescendantNodes().First(n => n is IdentifierNameSyntax identifier && identifier.Identifier.ContextualKind() == SyntaxKind.FieldKeyword));
+                Assert.Equal(SpecialType.System_Int32, typeInfo.Type.SpecialType);
+            }
+
+            comp.VerifyDiagnostics(
+                // (10,25): error CS8080: Auto-implemented properties must override all accessors of the overridden property.
+                //     public override int P1 { get { _ = field; return field; } }
+                Diagnostic(ErrorCode.ERR_AutoPropertyMustOverrideSet, "P1").WithLocation(10, 25),
+                // (10,25): error CS8080: Auto-implemented properties must override all accessors of the overridden property.
+                //     public override int P1 { get { _ = field; return field; } }
+                Diagnostic(ErrorCode.ERR_AutoPropertyMustOverrideSet, "P1").WithLocation(10, 25), // PROTOTYPE(semi-auto-props): Get rid of the extra diagnostic.
```

Comment:
I'm definitely planning to address this PROTOTYPE comment in this PR. Just taking suggestions on the general approach first.